deployments/chart/templates/clusterrole.yaml

```diff
@@ -0,0 +1,166 @@
+{{- if not .Values.constrainedDownscaler }}
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ include "go-kube-downscaler.fullname" . }}
+rules:
+- apiGroups:
+    - ""
+  resources:
+    - pods
+    - namespaces
+  verbs:
+    - get
+    - watch
+    - list
+- apiGroups:
+    - apps
+  resources:
+    - deployments
+    - statefulsets
+    - daemonsets
+  verbs:
+    - get
+    - watch
+    - list
+    - update
+    - patch
+- apiGroups:
+    - argoproj.io
+  resources:
+    - rollouts
+  verbs:
+    - get
+    - watch
+    - list
+    - update
+    - patch
+- apiGroups:
+    - autoscaling
+  resources:
+    - horizontalpodautoscalers
+  verbs:
+    - get
+    - watch
+    - list
+    - update
+    - patch
+- apiGroups:
+    - batch
+  resources:
+    - cronjobs
+  verbs:
+    - get
+    - watch
+    - list
+    - update
+    - patch
+- apiGroups:
+    - keda.sh
+  resources:
+    - scaledobjects
+  verbs:
+    - get
+    - watch
+    - list
+    - update
+    - patch
+- apiGroups:
+    - zalando.org
+  resources:
+    - stacks
+  verbs:
+    - get
+    - watch
+    - list
+    - update
+    - patch
+- apiGroups:
+    - ""
+  resources:
+    - events
+  verbs:
+    - get
+    - create
+    - watch
+    - list
+    - update
+    - patch
+- apiGroups:
+    - policy
+  resources:
+    - poddisruptionbudgets
+  verbs:
+    - get
+    - watch
+    - list
+    - update
+    - patch
+- apiGroups:
+  - constraints.gatekeeper.sh
+  resources:
+  - kubedownscalerjobsconstraint
+  verbs:
+  - get
+  - create
+  - watch
+  - list
+  - update
+  - patch
+  - delete
+- apiGroups:
+  - kyverno.io
+  resources:
+  - policies
+  resourceNames:
+  - kube-downscaler-jobs-policy
+  verbs:
+  - get
+  - create
+  - watch
+  - list
+  - update
+  - patch
+  - delete
+- apiGroups:
+  - kyverno.io
+  resources:
+  - policies
+  verbs:
+  - get
+  - create
+  - watch
+  - list
+- apiGroups:
+  - templates.gatekeeper.sh
+  resources:
+  - constrainttemplate
+  verbs:
+  - create
+  - get
+  - list
+  - watch
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - create
+  - get
+  - list
+  - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ include "go-kube-downscaler.fullname" . }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ include "go-kube-downscaler.fullname" . }}
+subjects:
+  - kind: ServiceAccount
+    name: {{ include "go-kube-downscaler.serviceAccountName" . }}
+    namespace: {{ .Release.Namespace }}
+{{- end }}
```

Comment:
This could be done, however this would create two unnecessary if checks which are already covered in the role files.

Reply:
I would still go ahead with this. An additional 'if-else' statement isn’t such a big deal

Reply:
I would rather prefer to leave them as is because it makes the roles and rolebindings consistent. If we split the rolebinding from the role creation we have to iterate over all the namespaces again seperately which is not too pretty.